### DISCUSSION TITLE
ente web "Critical dependency"

### DISCUSSION THREAD

**Discussion by roughnecks** [ente-io/ente#1426] [Q&A]
Hi there,
I am trying to self-host ente web (not from docker though) and I get a lot of warning when installing, like this:

```
warning "workspace-aggregator-f7348bbc-08cf-4097-97db-ca0a451f3cd3 > photos > react-virtualized-auto-sizer@1.0.23" has unmet peer dependency "r
eact-dom@^15.3.0 || ^16.0.0-alpha || ^17.0.0 || ^18.0.0".
```

when it finished I went ahead all the same and launched it:

```
$ NEXT_PUBLIC_ENTE_ENDPOINT=http://localhost:8080 yarn dev
yarn run v1.22.21
$ yarn dev:photos
$ yarn workspace photos next dev
$ /home/user/ente/web/node_modules/.bin/next dev
   ▲ Next.js 14.1.0
   - Local:        http://localhost:3000
   - Environments: .env.development, .env

 ✓ Ready in 5s
 ○ Compiling / ...
 ⚠ ../../node_modules/libheif-js/libheif-wasm/libheif-bundle.js
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted

Import trace for requested module:
../../node_modules/libheif-js/libheif-wasm/libheif-bundle.js
../../node_modules/libheif-js/wasm-bundle.js
../../node_modules/heic-decode/index.js
../../node_modules/heic-convert/index.js
./src/worker/convert.worker.ts
./src/utils/comlink/ComlinkConvertWorker.ts
./src/services/heic-convert/service.ts
./src/services/heicConversionService.ts
./src/utils/file/index.ts
./src/utils/export/index.ts
```

At first the output stopped at "Ready in 5s", but as soon as I opened the page in my browser (I'm reverse proxying it with nginx), it continued with the other messages.. I can see the login page just fine but not sure if all of this is to be expected.. On a side note, the web server is taking 1.2GB of RAM just to show the landing page.

**Comment by mnvr** [comment]
What's the problem you're facing?

yarn dev is for development. yarn build is for a production build. Please try to make an effort to go through the docs and/or code instead of asking at every step you're doing. I've already shared links to help pages, e.g. there is a this guide from someone who has run a similar setup - https://help.ente.io/self-hosting/guides/external-s3.

**Reply by roughnecks** [reply]
Sorry but I followed right here: https://help.ente.io/self-hosting/

**Reply by mnvr** [reply]
The slowness you're seeing is because you're doing a `yarn dev`. That runs a development server, which is going to be slower. It's is fine for setting things up, but if you are planning to use it for long, I would suggest doing a `yarn build`. This creates a production build, which is a static site consisting of a folder of HTML+CSS+JS files that can then be deployed on any standard web server. 

Nginx is a common choice for a web server. In the guide from the community member I linked to, they did a `yarn build` and copied the generate static site (from the `web/apps/photos/out` folder) to where nginx serves them. Note that the community guide is just one way, it's not the only way - the basic gist is that `yarn build` will produce a `web/apps/photos/out` folder that you can then serve with any web server of your choice.

**Reply by mnvr** [reply]
Another option you have is to use the dockerfile from this thread - https://github.com/ente-io/ente/discussions/1183 - that'll build an image with the web app that you can just run using docker without needing to setup nginx locally.

**Reply by roughnecks** [reply]
OK, I didn't know anything about that. Are you sure the link you posted is the right one? I mean https://help.ente.io/self-hosting/guides/external-s3 says absolutely nothing about the `web/apps/photos/out` folder to serve. It talks about building ente and web with docker, unless I'm totally blind.

**Reply by roughnecks** [reply]
So far I'm only testing the web app, not sure if I want to host it, so I would like not to run that in docker.

**Reply by roughnecks** [reply]
Sorry to bother you again but I'm having issues.
Logged in the web app and it synced alright. Also manually uploaded another folder and removed duplicates, everything seemed to work. Now I noticed that when opening *any* single picture I get this error: https://i.imgur.com/1sCdw3p.jpeg

**Reply by roughnecks** [reply]
And these are the last debug logs: https://gist.githubusercontent.com/roughnecks/f7a15ec009b5fbb46c5bdc05b612679b/raw/fe67f2cf9ce26823bb987482af4bc95b26d3660f/2024-04-12_17-48-32.txt
